encodings/fastlanes/src/bitpacking/compute/search_sorted.rs

```diff
@@ -71,27 +102,29 @@ impl BitPackedSearch {
                     .vortex_expect("Only sparse patches are supported")
                     .min_index()
             }),
-            validity: array.validity(),
+            first_null_idx,
         }
     }
 }
 
-impl<T: BitPacking + NativePType> IndexOrd<T> for BitPackedSearch {
+impl<T: BitPacking + NativePType> IndexOrd<T> for BitPackedSearch<'_, T> {
     fn index_cmp(&self, idx: usize, elem: &T) -> Option<Ordering> {
         if let Some(min_patch) = self.min_patch_offset {
             if idx >= min_patch {
                 return Some(Greater);
             }
         }
 
-        if self.validity.is_null(idx) {
+        // Null is always at the end, and the value we're searching for is non-null, thus if we
+        // see a null, always branch to the left.
+        if idx >= self.first_null_idx {
```

Comment:
is it better to have an if here or slice it upfront so there's no branch on each search?

Reply:
My intuition was that slice overhead is great than the overhead of an extra few comparisons but I did not measure that

Reply:
Probably fair. Let’s keep it as is and if we notice it we can improve